### PR TITLE
added new option to export query with placeholders

### LINKED ISSUE #95: Overpass-turbo and uMap
Hi :)

One of the most common requests I get from uMap support, is to ease the bridge between overpass-turbo and uMap.

uMap can eat OSM data (thanks to https://github.com/tyrasd/osmtogeojson, and so thanks to you :) ), and for this one way is to use an Overpass URL.

What I usually answer to users when they want to export data to uMap is to download as geojson from overpass-turbo, and then import the file in uMap (or copy/paste it).

But uMap also allows the data to be fetched dynamically when a map is loaded. This is called "remote data" in uMap. Roughly, one creates a new layer, and set a "remote data" URL. So now each time the map is loaded, the URL is requested, and the content shown on the map (one can style the data even if it's remotely fetched). The goal is to have fresh data. See for example: http://umap.openstreetmap.fr/en/map/pere-lachaise_6765#16/48.8613/2.3953

The Overpass data URL is (unless I've missed something :) ) a bit hidden. What I usually do to get it is: export => query => click on compact => right click on the link in the new page.

That said, I see two ways to create a better bridge between Overpass-turbo and uMap:
- Make it easier to get the Overpass data URL: for example, adding a section on "Export" called "URL" where the Overpass data URL can be copy-pasted
- this one is a bit more specific: on "Export" => "Data", adding a "Create a custom map" new link, that points to http://umap.openstreetmap.fr/map/new/?importUrl={overpass-url}&importFormat=osm (that URL would be obviously configurable: for example, Swiss instance of overpass-turbo would point to http://umap.osm.ch instead).

Those are just suggestions to start a discussion. I may have missed important points, and maybe not every points of mines make sense (I'm not sure for example how overpass-turbo deals with Overpass "convert" step) ;)

Thanks in advance for your thoughts on this!

Yohan

### PR DESCRIPTION
**Pull request by ANGkeith**
closes #95 
screenshot from 2018-10-05 03-27-44

### CODE REVIEW

File: js/ide.js [hunk 1]
```diff
@@ -2127,6 +2127,29 @@ var ide = new function() {
         encodeURIComponent(query) +
         "&target=compact";
 
+      function constructOverpassQLUrl(query) {
+        // remove /* */ comments from query
+        query = query.replace(/\/\*(.|\n)*?\*\//g, "");
```
Comment (tyrasd) [inline js/ide.js]: I believe using `[\S\s]` (i.e. _whitespace_ or _not whitespace_) is better to match multiline strings in javascript (your approach wouldn't work with `\r` newlines or other exotic stuff that doesn't match the `.` in js. see https://stackoverflow.com/a/1981692/1627467 or https://www.regular-expressions.info/dot.html for further info

File: js/ide.js [hunk 2]
```diff
@@ -2127,6 +2127,29 @@ var ide = new function() {
...
+        // replace bbox with south west north east
+        query = query.replace(/{{bbox}}/g, "{south},{west},{north},{east}");
+        // replace //  comments from query
+        query = query.replace(/\/\/.*/g, "");
+        // removes indentation
+        query = query.replace(/\n(\s|\t)+/g, "");
```
Comment (tyrasd) [inline js/ide.js]: doesn't `\s` also match tabulators? so simply `replace(/\n\s+/g, "")` would be enough, right? Also, this replace could be combined with the next line by writing `replace(/\n\s*/g, "")`, right?

**Comment by ANGkeith** [comment]
ok i will make the respective changes on monday

File: js/ide.js (hunk 1 above)
Comment (ANGkeith) [reply js/ide.js]: I didn't knew about \r! thanks

**Comment by tyrasd** [comment]
I've just merged this (and the additional changes in #400) in #402 where I additionally tweaked a few things (see https://github.com/tyrasd/overpass-turbo/pull/402#issue-224527495). You can test the final thing here: http://overpass-turbo.eu/master/

**Comment by sfkeller** [comment]
@tyrasd : Sorry coming back to this issue again: It seems that the merge did not succeed for http://overpass-turbo.eu/ . The only place I see the patch of @ANGkeith (and me) is at http://overpass-turbo.osm.ch/ given browser chooses the english version. 

So, there seems to be two issues: First the (non-)deployment at overpass-turbo.eu, and second the fact that it's not showing up in the german version. 

Can you look at this once again pls.?

**Comment by sfkeller** [comment]
@tyrasd Hallo Martin: Kannst du diesem eigentlich schon gemergden Pull-Request nochmals einen Schubs geben? Er scheint nicht im deployten Service angekommen zu sein.

Dann ist da noch die Transifex-Übersetzung. Diese ist offenbar mind. auf Deutsch parat. Was mir/uns da noch fehlt ist das Verständnis, wie die Übersetzungen mit dem Source Code zusammen spielen...

**Pull request merged**